ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/feemarket/FeeMarketBuilderFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet.feemarket;
+
+import org.hyperledger.besu.config.BlobSchedule;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpecBuilder;
+
+import java.util.Optional;
+
+/**
+ * Factory for creating {@link FeeMarket} instances used in protocol specification configuration.
+ */
+public class FeeMarketBuilderFactory {
+  @FunctionalInterface
+  public interface FeeMarketBuilder {
```

Comment:
nit: It's little confusing having this as well as ProtocolSpecBuilder.FeeMarketBuilder sharing the same name. 
IIUC, they are working together just at two different layers of abstraction.

The outer FeeMarketBuilder accepts only blobSchedule and returns FeeMarket.
The inner represents one case where we delegate to the "inner" builder which accepts more params and return FeeMarket.

I feel like both are more accurately "Factorys" rather than "Builders", but since the outer is following an existing convention I would suggest we change this one to be: `FeeMarketFactory`
Perhaps this interface should also live where the static factories live, i.e. in FeeMarket?